nixos/modules/services/games/pterodactyl-wings.nix

```diff
@@ -0,0 +1,214 @@
+      '';
+    };
+    settings = mkOption {
+      description = ''
+        Configuration options for the wings daemon.
+        Some required values must be generated from the controlling panel.
+        See [reference](https://github.com/pterodactyl/wings/blob/develop/config/config.go)
+        for all options.
+      '';
+      default = {};
+      example = {
+        remote = "https://example.com";
+        uuid = "e851b74c-2e88-4d5c-8c6e-381218c7119d";
+        token_id = "fdZUuYH0k4aa8OGJ";
+        api = {
+          host = "127.0.0.1";
+          port = 6868;
+          upload_limit = 1000;
+        };
+        system = {
+          sftp.bind_port = 6869;
+          root_directory = "/srv/pterodactyl";
+          data = "/srv/pterodactyl/volumes";
+          archive_directory = "/srv/pterodactyl/archives";
+          backup_directory = "/srv/large/pterodactyl/backups";
+        };
+       allowed_origins = [ "https://panel.example.com" ];
+      };
+      type = submodule {
+        freeformType = settingsFormat.type;
+        options = {
+          uuid = mkOption {
+            type = str;
+            description = ''
+              A unique identifier for this node in the Panel.
+              Input this from the configuration generated on the panel.
+            '';
+          };
+          token_id = mkOption {
+            type = str;
+            description = ''
+              An identifier for the token which must be included in any requests
+              to the panel so that the token can be looked up correctly.
+              Input this from the configuration generated on the panel.
+            '';
+          };
+          api = {
+            host = mkOption {
+              type = str;
+              default = "0.0.0.0";
+              example = "[::1]";
+              description = "The interface that the internal webserver should bind to.";
+            };
+            port = mkOption {
+              type = port;
+              default = 8080;
+              description = "The port that the internal webserver should bind to.";
+            };
+            ssl = {
+              enabled = mkEnableOption "HTTPS for the API interface";
+              cert = mkOption {
+                type = nullOr path;
+                default = null;
+                description = "Path to certificate to use for HTTPS.";
+              };
+              key = mkOption {
+                type = nullOr path;
+                default = null;
+                description = "Path to key to use for HTTPS.";
+              };
+            };
+          };
+          system = {
+            root_directory = mkOption {
+              type = path;
+              default = "/var/lib/pterodactyl";
+              description = "Root directory where daemon data is stored.";
+            };
+            log_directory = mkOption {
+              type = path;
+              default = "/var/log/pterodactyl";
```

Comment:
We cannot log to journald?

Reply:
The daemon also logs to stdout, and so journald captures the output. I didn't see a way to disable file logging, so I left the option.

Reply:
Can we redirect the file logging to /dev/log which ends up in journald?

Reply:
I think that would end up double-logging because it would duplicate the stdout logging. The daemon also logs to arbitrary files in the directory, which I'm not sure how to redirect.